Automattic/wp-calypso
# Jetpack Cloud: Allow restore using staging-* role

## Proposed Changes

* No UI changes
* Allow search and selection of sites with `staging-*` credential role from the destination search box.
* Should be able to restore using selected site with `staging-*` role.
* Record `calypso_jetpack_clone_flow_set_staging_site_with_role` tracks event on selection of a site with `staging-*` role.

## Testing Instructions



* Save a staging credential following https://github.com/Automattic/wp-calypso/pull/83035 test steps
* Use the generated Jetpack/Calypso live link from this PR to search the site added above.
* Verify the destination URL is properly displayed underneath _Select a point to copy / Configure your copy_ scene after the site selection in the above step.
  <img width="567" alt="Screenshot 2023-10-25 at 5 10 57 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/bfb9caf8-0b59-4535-9207-b6fdd3eceb30">
* Proceed to restore the site using the selected staging credentials site, it should be completed. 

## Pre-merge Checklist
